Source/Thirdweb/Private/ThirdwebSubsystem.cpp

```diff
@@ -0,0 +1,178 @@
+// Copyright (c) 2024 Thirdweb. All Rights Reserved.
+
+// ReSharper disable CppUE4BlueprintCallableFunctionMayBeStatic
+
+#include "ThirdwebSubsystem.h"
+#include "HttpServerModule.h"
+#include "IHttpRouter.h"
+#include "Thirdweb.h"
+#include "ThirdwebCommon.h"
+#include "ThirdwebLog.h"
+#include "ThirdwebRuntimeSettings.h"
+#include "ThirdwebWalletHandle.h"
+#include "Engine/StreamableManager.h"
+#include "Kismet/GameplayStatics.h"
+#include "Misc/DefaultValueHelper.h"
+
+void UThirdwebSubsystem::Initialize(FSubsystemCollectionBase& Collection)
+{
+	Super::Initialize(Collection);
+	Settings = GetDefault<UThirdwebRuntimeSettings>();
+	TW_LOG(Log, TEXT("ThirdwebSubsystem::Intitialize::Subsystem initialization complete"))
+}
+
+void UThirdwebSubsystem::Deinitialize()
+{
+	Super::Deinitialize();
+	TW_LOG(Log, TEXT("ThirdwebSubsystem::Deinitialize::Subsystem deinitialization complete"))
+}
+
+EFunctionResult UThirdwebSubsystem::BP_CreateInAppEmailWallet(const FString& Email, FWalletHandle& Wallet, FString& Error)
+{
+	return CreateInAppEmailWallet(Email, Wallet, Error) ? EFunctionResult::Success : EFunctionResult::Failed;
+}
+
+bool UThirdwebSubsystem::CreateInAppEmailWallet(const FString& Email, FWalletHandle& Wallet, FString& Error)
+{
+	if (Thirdweb::create_in_app_wallet(Settings->GetClientID(), StringCast<ANSICHAR>(*Email).Get(), Settings->GetSecretKey(), StringCast<ANSICHAR>(*Email).Get(), Settings->GetStorageDirectoryPath(),
+	                                   nullptr).AssignResult(Error))
+	{
+		Wallet = FWalletHandle(FWalletHandle::InApp, Error);
+		Error.Empty();
+		return true;
+	}
+	return false;
+}
+
+EFunctionResult UThirdwebSubsystem::BP_CreateInAppOAuthWallet(const EThirdwebOAuthProvider Provider, FWalletHandle& Wallet, FString& Error)
+{
+	return CreateInAppOAuthWallet(Provider, Wallet, Error) ? EFunctionResult::Success : EFunctionResult::Failed;
+}
+
+bool UThirdwebSubsystem::CreateInAppOAuthWallet(const EThirdwebOAuthProvider Provider, FWalletHandle& Wallet, FString& Error)
+{
+	if (Thirdweb::create_in_app_wallet(Settings->GetClientID(), Settings->GetBundleID(), Settings->GetSecretKey(), nullptr, Settings->GetStorageDirectoryPath(),
+	                                   StringCast<ANSICHAR>(*ToString(Provider)).Get()).AssignResult(Error))
+	{
+		Wallet = FWalletHandle(FWalletHandle::InApp, Error);
+		Error.Empty();
+		return true;
+	}
+	return false;
+}
+
+
+EFunctionResult UThirdwebSubsystem::BP_CreateSmartWallet(const FWalletHandle& LocalWallet, FWalletHandle& SmartWallet, FString& Error, const int64 ChainID, const bool bGasless, const FString& Factory,
```

Comment:
Can we call this PersonalWallet - to avoid confusion with our previous product LocalWallet which is equivalent to PrivateKeyWallet. A Smart Wallet can take in any type of wallet (handle) and work.